[doc] Improve Pulsar Security-token-admin (#5052)
apache/pulsar: site2/docs/security-token-admin.md

## Token Authentication Overview

Pulsar supports authenticating clients using security tokens that are based on [JSON Web Tokens](https://jwt.io/introduction/) ([RFC-7519](https://tools.ietf.org/html/rfc7519)).

You can use tokens to identify a Pulsar client and associate with some "principal" (or "role") that
is permitted to do some actions (for example, publish to a topic or consume from a topic).

> A user typically gets a user a token string from the administrator (or some automated service).

Reviewer: check this sentence.

The compact representation of a signed JWT is a string that looks like as the follwing:

```
eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJKb2UifQ.ipevRNuRP6HflG8cFKnmUPtypruRC4fb1DWtoLL62SY
```

> Application specifies the token when you are creating the client instance. An alternative is to pass a "token supplier" (a function that returns the token when the client library needs one).

Reviewer: Check the first sentence, and pay attention to two points:
1. the word "specifies"
2. use present tense.

> #### Always use TLS transport encryption
> Sending a token is equivalent to sending a password over the wire. You had better

> > use TLS encryption all the time when you are talking to the Pulsar service. See

Reviewer: replace "are talking to" to other words.

> [Transport Encryption using TLS](security-tls-transport.md) for more details.

## Secret vs Public/Private keys

> JWT support two different kind of keys in order to generate and validate the tokens:

Reviewer: JWT supports...two...kinds...

 * Symmetric :
    - You can use a single ***Secret*** key to generate and validate tokens.
 * Asymmetric: A pair of keys consist of the Private key and the Public key.
    - You can use ***Private*** key to generate tokens
    - You can use ***Public*** key to validate tokens

### Secret key

> When you are using a secret key, the administrator creates the key and uses the key to generate the client tokens. You can also configure this key to the brokers in order to allow the brokers validating the clients.

Reviewer: Use present tense, for example, "when you use a xxx". Check all similar cases.

Pulsar supports authenticating clients using security tokens that are based on [JSON Web Tokens](https://jwt.io/introduction/) ([RFC-7519](https://tools.ietf.org/html/rfc7519)).

You can use tokens to identify a Pulsar client and associate with some "principal" (or "role") that is permitted to do some actions (for example, publish to a topic or consume from a topic).

> A user typically gives a user a token string from the administrator (or some automated service).

Reviewer: A user typically gets a token string from the administrator (or some automated service).

The compact representation of a signed JWT is a string that looks like as the follwing:

```
eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJKb2UifQ.ipevRNuRP6HflG8cFKnmUPtypruRC4fb1DWtoLL62SY
```

> Application does the token specification when you create the client instance. An alternative is to pass a "token supplier" (a function that returns the token when the client library needs one).

Reviewer: re-write this sentence. 
The token is specified by application when you create a client instance.

## Authorization

The token does not associate any permission. The authorization engine determines whether the token has permissions or not. Once you have created the token, you can grant permission for this token to do certain actions. For example:

```shell
$ bin/pulsar-admin namespaces grant-permission my-tenant/my-namespace \
            --role test-user \
            --actions produce,consume
```

> ## Enable token authentication

Reviewer: Refine it further.

## Authorization

> The token itself does not have any permission associated. The authorization engine determines whether the token should have permissions or not. Once you have created the token, one can grant permission for this token to do certain actions. For example:

Reviewer: ```suggestion
The token does not associate any permission. The authorization engine determines whether the token has permissions or not. Once you have created the token, you can grant permission for this token to do certain actions. For example:
```

```shell
$ bin/pulsar-admin namespaces grant-permission my-tenant/my-namespace \
            --role test-user \
            --actions produce,consume
```

> ## Enable Token Authentication ...

Reviewer: Refine the titles, you can refer to https://github.com/apache/pulsar/pull/5038
